### PR TITLE
[Zero-Dim] support matmul/ReshapeTransform/nll_loss to support 0D

### PR DESCRIPTION
**Pull request by zhwesky2010**
### PR types

New features
### PR changes

APIs

### Description

Pcard-66984

修复以下3种case输出0D：
```
1. paddle.distribution.ReshapeTransform的_forward_log_det_jacobian类函数
2. paddle.nn.functional.nll_loss在reduce为sum或mean时
3. paddle.matmul在输入x, y均为1D Tensor时，为向量点积(标量积)，应输出0D Tensor表示标量
```

### CODE REVIEW

**Review by jeff41404 — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]
LGTM
